app/src/main/java/com/x8bit/bitwarden/ui/platform/components/dialog/BitwardenPinDialog.kt

```diff
@@ -0,0 +1,78 @@
+package com.x8bit.bitwarden.ui.platform.components.dialog
+
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.testTag
+import androidx.compose.ui.semantics.testTagsAsResourceId
+import com.x8bit.bitwarden.R
+import com.x8bit.bitwarden.ui.platform.components.button.BitwardenTextButton
+import com.x8bit.bitwarden.ui.platform.components.field.BitwardenPasswordField
+
+/**
+ * Represents a Bitwarden-styled dialog for the user to enter their PIN.
+ *
+ * @param onConfirmClick called when the confirm button is clicked and emits the entered PIN.
+ * @param onDismissRequest called when the user attempts to dismiss the dialog (for example by
+ * tapping outside of it).
+ */
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun BitwardenPinDialog(
+    onConfirmClick: (masterPassword: String) -> Unit,
+    onDismissRequest: () -> Unit,
+) {
+    var pin by remember { mutableStateOf("") }
```

Comment:
💭 should this be saveable, the edge case of someone entering `**` -> `change to dark mode` and they would lose the values entered. Not sure if we are specifically not doing this as security thing.

Reply:
Excellent question - I actually don't know so I'll defer to @david-livefront and @SaintPatrck for the security aspect of that question.

Reply:
As a user, if my theme changes while this dialog is displayed I would not expect to lose my input. Beyond a user manually changing it, but still an edge case, the theme could change if it's based on time of day. 

@david-livefront this is reminiscent of the issues you worked on with Dillon regarding input being retained for too long. Any concerns about that here?

Reply:
I would expect us to not save this for security reasons, I believe we don't use `remeberSavable` in the `BitwardenMasterPasswordDialog` either.

It is annoying, the theme is an edge-case but rotation will also clear the state and we just have to live with that.